lib/ansible/modules/utilities/logic/include_role.py

```diff
@@ -20,16 +20,15 @@
 module: include_role
 short_description: Load and execute a role
 description:
-  - Loads and executes a role as a task dynamically. This frees roles from the `roles:` directive and allows them to be
-    treated more as tasks.
-  - Unlike M(import_role), most keywords, including loop, with_items, and conditionals, apply to this statement.
+  - When used underneath a `tasks:` directive, M(include_role) dynamically loads and executes a specified role as a task.
```

Comment:
this phrasing is implying you can use in places outside a `tasks:` directive, which might be more confusing to users.

It can only be used where 'tasks' are allowed pre_tasks/tasks/post_tasks and inside a role

not to be confused with `import_playbook` which is really 'only task outside of a task container' .. since it is not really a task.

Reply:
> this phrasing is implying you can use in places outside a tasks: directive, which might be more confusing to users.

You CAN. Try the code snippet below: it's exactly why I WAS confused, because I used it in a roles: section, having tagged all tasks in the role (and the included role), and it happily accepted my use of include_roles that way. You can see why - it's reminiscent of how one can call a role using a tag. The only thing it didn't do is use a list of dicts inside the role with `item` nomenclature, but the syntax was accepted.

If the fact that the syntax is allowed is a bug, then I'll lodge one, but I hedged my wording because it actually is allowed in a roles directive. I just had no idea if it had some other purpose. What you're saying is 'it doesn't and shouldn't be allowed/encouraged anyway'.

```
- hosts: f5
  connection: local
  roles:
    - include_role:
      name: f5_commands
      apply:
        tags:
          - f5_commands
      tags:
        - always
      loop: "{{ list_of_dicts }}"
```

Reply:
(I'd like to confirm your answer to 'You should never do that' before I make a change to how I word the description in include_tasks - but there was a reason I said 'When used this way', not 'This should only be used in a tasks: block'.)

Reply:
Ping: @bcoca , confirm whether the ability to use this outside of a tasks: block is a code implementation error (i.e., that it should not be attempted in Ansible code regardless of if it's technically possible)